### PR TITLE
git: repo: base: update clone_from env argument description

### PR DESCRIPTION
**Pull request by pared**
Related to #924, https://github.com/iterative/dvc/issues/2471

### CODE REVIEW

**Comment by pared** [comment]
@Byron please take a look, when able.

File: git/repo/base.py [hunk 1]
```diff
@@ -1012,6 +1012,9 @@ def clone_from(cls, url, to_path, progress=None, env=None, multi_options=None, *
         :param to_path: Path to which the repository should be cloned to
         :param progress: See 'git.remote.Remote.push'.
         :param env: Optional dictionary containing the desired environment variables.
+            Note: Provided variables will be used to update the execution
+            environment for `git`. If some variable is not specified in `env`
```
Comment (Byron) [inline git/repo/base.py]: Maybe you could also hint at how to 'unset' environment variables that way.
It's just a suggestion though, I am happy to merge as is.

Comment (pared) [reply git/repo/base.py]: Sure, I'll mention that :)

**Comment by Byron** [comment]
Perfect, thanks so much!

**Pull request merged**